src/AutoRemoteSyscalls.cc

```diff
@@ -255,6 +256,16 @@ long AutoRemoteSyscalls::syscall_base(int syscallno, Registers& callregs) {
         // We entered the syscall, so stop now
         break;
       }
+      if (t->ptrace_event() == PTRACE_EVENT_EXIT) {
+        // We died, just let it be
+        break;
+      }
+      if (t->stop_sig() == SIGTRAP && t->get_siginfo().si_code == TRAP_TRACE) {
```

Comment:
If this is AArch64 specific, perhaps it should be ifdefed?

Reply:
I think powerpc has the same behavior, so it might just be x86 that's the odd one out. I don't think It'll do any harm. That said, I am actually a bit confused why TRAP_TRACE doesn't seem to show up for us on x86, because it seems like it should:
```
static inline int get_si_code(unsigned long condition)
{
	if (condition & DR_STEP)
		return TRAP_TRACE;
	else if (condition & (DR_TRAP0|DR_TRAP1|DR_TRAP2|DR_TRAP3))
		return TRAP_HWBKPT;
	else
		return TRAP_BRKPT;
}
[snip]
	si_code = get_si_code(tsk->thread.debugreg6);
	if (tsk->thread.debugreg6 & (DR_STEP | DR_TRAP_BITS) || user_icebp)
		send_sigtrap(regs, error_code, si_code);
```
I do know why we don't see it here, which is that aarch64 reports an extra single step trap after exiting the syscall, but not sure why we don't see it in general (or maybe we do, and I just never looked closely enough).